### PR TITLE
add sinh/cosh/tanh support for llvm19+

### PR DESCRIPTION
**Pull request by ZuseZ4**

### CODE REVIEW

File: enzyme/tools/enzyme-tblgen/enzyme-tblgen.cpp [hunk 1]
```diff
@@ -2632,6 +2637,11 @@ void emitDiffUse(const RecordKeeper &recordKeeper, raw_ostream &os,
       break;
     }
     case IntrDerivatives: {
+      auto minVer = pattern->getValueAsInt("minVer");
```
Comment (wsmoses) [inline enzyme/tools/enzyme-tblgen/enzyme-tblgen.cpp:2640]: See above, there is prior support for passing min/max versions

**Review by wsmoses — CHANGES_REQUESTED** [review]

File: enzyme/tools/enzyme-tblgen/enzyme-tblgen.cpp (hunk 1 above)
Comment (ZuseZ4) [reply enzyme/tools/enzyme-tblgen/enzyme-tblgen.cpp]: done

**Comment by ZuseZ4** [comment]
@wsmoses I'll claim those failures as not caused by me

**Comment by wsmoses** [comment]
hm they don't fail on current main tho

File: enzyme/Enzyme/InstructionDerivatives.td [hunk 2]
```diff
@@ -374,7 +374,7 @@ def : CallPattern<(Op $x),
                   >;
 
 def : CallPattern<(Op $x),
-                  ["sinh"],
+                  ["sinh", "", "18"],
```
Comment (wsmoses) [inline enzyme/Enzyme/InstructionDerivatives.td:377]: oh yeah this isn't valid here

Comment (wsmoses) [reply enzyme/Enzyme/InstructionDerivatives.td:377]: so this is indeed the cause of the issues, I presume

File: enzyme/test/Enzyme/ReverseMode/cosh19.ll [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+; Function Attrs: nounwind readnone uwtable
+define double @tester(double %x) {
+entry:
+  %0 = call double @llvm.cosh.f64(double %x)
+  ret double %0
+}
+
+define double @test_derivative(double %x) {
+entry:
+  %0 = tail call double (double (double)*, ...) @__enzyme_autodiff(double (double)* nonnull @tester, double %x)
```
[the remark this answers is not in the document]
Comment (ZuseZ4) [reply enzyme/test/Enzyme/ReverseMode/cosh19.ll]: sg.

**Pull request merged**